### ISSUE TITLE
Butterknife 10.0.0 and Feature plugin do not work

### ISSUE THREAD

**Issue by frogermcs** [JakeWharton/butterknife#1441]
I mentioned it here: https://github.com/JakeWharton/butterknife/pull/1402 but MR is already closed.

In project with `com.android.feature`, the location of R.txt is a bit different, what leads to an error 

```
> java.io.FileNotFoundException: .../my_app/features/{module}/build/intermediates/symbols/feature/debug/R.txt (No such file or directory)
```

It seems that Gradle Plugin 3.3 puts this file into:
.../my_app/features/{module}/build/intermediates/symbols/debug/R.txt
(no /feature/ part)

**Comment by JakeWharton** [comment]
Are you using the feature plugin?

**Comment by frogermcs** [comment]
Yes we use ```apply plugin: 'com.android.feature'``` instead of library.

**Comment by JakeWharton** [comment]
Sorry I wasn't replying to you. That was in the original report and is presumably the source of the trouble.

**Comment by TAGASK** [comment]
Duplicate of #1451

**Comment by JakeWharton** [comment]
I think you have that backwards

**Comment by TAGASK** [comment]
After several tests, it turned out that this error occurs when the R2 file generation is used in the **baseFeature** and in an **instant app feature** at the same time.
I ll advice to move all the UI codebase which needs the R2 class file from the the baseFeature to the specific module (Instant app feature) you are using.
Hope this ll help.

**Comment by frogermcs** [comment]
Hey, while it can be great workaround, it's not that simple in our case. The multimodule architecture becomes pretty complex and there are UI components modules reused in higher level ones.

**Comment by TAGASK** [comment]
Anyway it doesn't come from the butterknife plugin, if there is a failure it s because the file is not created previously :
/build/intermediates/symbols/feature/debug/R.txt (No such file or directory)
This file is not generated by butterKnife, so it's in feature or baseFeature that there is an issue.

**Comment by frogermcs** [comment]
This file is generated in other directory like I mentioned in first message.

Gets this one:
```
* What went wrong:
Execution failed for task ':features:transaction:generateDebugFeatureR2'.
> java.io.FileNotFoundException: ...dev/jenkins/workspace/d-app_feature_migrate_AS_33-LM7AGHCBO6KD7O3WL344QWYUEFXRSWGFGB6I23T74ZK4WV7FCPVA/features/transaction/build/intermediates/symbols/feature/debug/R.txt (No such file or directory)
```

While R.txt is there, but in different dir:
```
.../features/transaction/build/intermediates/symbols/debug/R.txt
```

**Comment by TAGASK** [comment]
` .../features/transaction/build/intermediates/symbols/debug/R.txt`

this one contains the res id and layout of the baseFeature with all the resource of the dependecies.

`java.io.FileNotFoundException: ...dev/jenkins/workspace/d-app_feature_migrate_AS_33-LM7AGHCBO6KD7O3WL344QWYUEFXRSWGFGB6I23T74ZK4WV7FCPVA/features/transaction/build/intermediates/symbols/feature/debug/R.txt`
this one should contains the id and layout of the res of the feature only.

Use only the `symbol/debug/R.txt` ll not resolve the issue.

**Comment by frogermcs** [comment]
Ahh I see...

Also I recreated this problem in much simpler project.

Here's the branch with code: https://github.com/frogermcs/MultiModuleGithubClient/tree/update_to_AS_3.3
And output from the build: https://travis-ci.com/frogermcs/MultiModuleGithubClient/jobs/179287644

Interesting fact is that in this chain of features:
repositories -> repository -> base only the middle one doesn't have /symbols/feature/debug/R.txt 🤔

**Comment by frogermcs** [comment]
Posted issue into Tracker -> https://issuetracker.google.com/issues/125166026

**Comment by frogermcs** [comment]
Early tests of the fix in gradle plugin 3.5.0 alpha12 show that it may be fixed now. :) 

Working code:
https://github.com/frogermcs/MultiModuleGithubClient/tree/update_to_AS_3.3

Working build process:
https://travis-ci.com/frogermcs/MultiModuleGithubClient/builds/109270546

**Comment by JakeWharton** [comment]
Great! I'm going to preemptively close. Let me know if anything changes in the future.

**Comment by elksa** [comment]
> Early tests of the fix in gradle plugin 3.5.0 alpha12 show that it may be fixed now. :)
> 
> Working code:
> https://github.com/frogermcs/MultiModuleGithubClient/tree/update_to_AS_3.3
> 
> Working build process:
> https://travis-ci.com/frogermcs/MultiModuleGithubClient/builds/109270546

So exactly which was the solution, thanks!
